modules/host/storagefolders.go

```diff
@@ -0,0 +1,594 @@
+// but only accounts for the actual raw data. Sia also places a nontrivial
+// amount of load on the filesystem, potentially to the tune of millions of
+// files. These files have long, cryptographic names and may take up as much as
+// a gigabyte of space in filesystem overhead, depending on how the filesystem
+// is architected. The host is programmed to gracefully handle full disks, so
+// wihle it might cause the user surprise that the host can't break past 99%
+// utilization, there should not be any issues if the user overestimates how
+// much storage is available in the folder they have offered to Sia. The host
+// will put the drive at 100% utilization, which may cause performance
+// slowdowns or other errors if non-Sia programs are also trying to use the
+// filesystem. If users are experiencing problems, having them set the storage
+// folder size to 98% of the actual drive size is probably going to fix most of
+// the issues.
+//
+// 'SizeRemaining' is a variable that remembers how much storage is remaining
+// in the storage folder. It is managed manually, and is updated every time a
+// sector is added to or removed from the storage folder. Because there is no
+// property that inherently guarantees the correctness of 'SizeRemaining',
+// implementation must be careful to maintain consistency.
+//
+// The UID of the storage folder is a small number of bytes that uniquely
+// identify the storage folder. The UID is generated randomly, but in such a
+// way as to guarantee that it will not collide with the ids of other storage
+// folders. The UID is used (via the uidString function) to determine the name
+// of the symlink which points to the folder holding the data for this storage
+// folder.
+type storageFolder struct {
+	Size          uint64
+	SizeRemaining uint64
+	UID           []byte
+}
+
+// uidString returns the string value of the storage folder's UID. This string
+// maps to the filename of the symlink that is used to point to the folder that
+// holds all of the sector data contained by the storage folder.
+func (sf *storageFolder) uidString() string {
+	if len(sf.UID) != storageFolderUIDSize {
+		build.Critical("sector UID length is incorrect - perhaps the wrong version of Sia is being run?")
+	}
+	return hex.EncodeToString(sf.UID)
+}
+
+// emptiestStorageFolder takes a set of storage folders and returns the storage
+// folder with the lowest utilization by percentage. 'nil' is returned if there
+// are no storage folders provided with sufficient free space for a sector.
+//
+// Refusing to return a storage folder that does not have enough space prevents
+// the host from overfilling a storage folder.
+func emptiestStorageFolder(sfs []*storageFolder) (*storageFolder, int) {
+	lowestUtilization := float64(2) // Set higher than the max of 1 to protect against floating point imprecision.
+	winningIndex := -1              // Set to impossible value to prevent unintentionally returning the wrong storage folder.
```

Comment:
why use an index instead of a `*storageFolder`?
I think that would allow you do remove the `winner` boolean too. You check at the end if `winningFolder == nil`